src/scripts.js

```diff
@@ -53,31 +53,12 @@ const displayUserStepGoal = () => {
 }
 
 const displayUserFriends = () => {
-  const userFriendNames = user.friends
-  .map((friend) => {
-
-    return userRepository.data[friend].name;
-  })
-  .forEach((friend) => {
-    // console.log(friend)
-    // console.log(friend.split(' ')[0])
-    friend.split(' ')[0];
-  })
-  return userFriendNames;
-  // console.log(userFriendNames)
-
-//   const userFriendNames = user.friends.filter((friend) =>
-//     // console.log(userRepository.data[friend].name)
-//   friend === userRepository.data[friend].id;
-// ).map(friend) => {
-//   console.log(friend);
-//   return friend.name
-// })
-//   // console.log(userFriendNames);
-//   return userFriendNames;
-//   userFriends.innerText = `${userRepository.data[friend].name}`;
+  const friends = userRepository.getUsersByIds(user.friends);
+  const friendNames = friends.map((friend) => {
+    return friend.name;
+  });
+  userFriends.innerText = `${friendNames.join(', ')}`;
```

Comment:
We pass in the array of the friend ids to the UserRepository method. That returns an array of those objects from the data. Then we map those objects to an array of name strings. Finally we join that array of strings with a comma and a space and display it on the DOM.